File: fetch_notion_posts.py
import requests
import os
from datetime import datetime

NOTION_API_TOKEN = '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'
DATABASE_ID = 'f3b3b3b3-7b7b-4b4b-8b8b-9b9b9b9b9b9b'
HEADERS = {
    "Authorization": f"Bearer {NOTION_API_TOKEN}",
    "Content-Type": "application/json",
    "Notion-Version": "2022-06-28"
}

def fetch_notion_posts():
    url = f"https://api.notion.com/v1/databases/{DATABASE_ID}/query"
    response = requests.post(url, headers=HEADERS)
    data = response.json()
    
    for result in data['results']:
        title = result['properties']['Title']['title'][0]['text']['content']
        category = result['properties']['Category']['select']['name']
        content = result['properties']['Content']['rich_text'][0]['text']['content']
        slug = result['properties']['Slug']['rich_text'][0]['text']['content']
        date = result['properties']['Date']['date']['start']
        
        md_content = f"""---
layout: post
title: "{title}"
date: {date}
category: {category}
slug: {slug}
---

{content}
"""
        filename = f"_posts/{date}-{slug}.md"
        with open(filename, 'w') as file:
            file.write(md_content)

if __name__ == "__main__":
    fetch_notion_posts()

File: fetch_notion_posts.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');
require('dotenv').config();

const NOTION_API_TOKEN = process.env.NOTION_API_TOKEN;
const DATABASE_ID = process.env.DATABASE_ID;

const headers = {
  'Authorization': `Bearer ${NOTION_API_TOKEN}`,
  'Content-Type': 'application/json',
  'Notion-Version': '2022-06-28'
};

async function fetchNotionPosts() {
  const url = `https://api.notion.com/v1/databases/${DATABASE_ID}/query`;
  try {
    const response = await axios.post(url, {}, { headers });
    const data = response.data;

    // Print the raw data to see the response from Notion
    console.log("Raw data from Notion:");
    console.log(JSON.stringify(data, null, 2));

    data.results.forEach(result => {
      const title = result.properties.Title?.title?.[0]?.text?.content || 'Untitled';
      const category = result.properties.Category?.rich_text?.[0]?.text?.content || 'Uncategorized';
      const contentArray = result.properties.Content?.rich_text || [];
      const content = contentArray.map(item => item.text.content).join('\n') || '';
      const slug = result.properties.Slug?.rich_text?.[0]?.text?.content || title.toLowerCase().replace(/ /g, '-');
      const date = result.properties.Date?.date?.start || new Date().toISOString().split('T')[0];

      // Print each post's details to verify the fetched data
      console.log(`Title: ${title}`);
      console.log(`Category: ${category}`);
      console.log(`Content: ${content}`);
      console.log(`Slug: ${slug}`);
      console.log(`Date: ${date}`);
      console.log("-----");

      const mdContent = `---
layout: post
title: "${title}"
date: ${date}
category: ${category}
slug: ${slug}
---

${content}
`;
      const categoryFolder = path.join(__dirname, `_posts`, category.toLowerCase().replace(/ /g, '-'));
      if (!fs.existsSync(categoryFolder)) {
        fs.mkdirSync(categoryFolder, { recursive: true });
      }
      const filename = path.join(categoryFolder, `${date}-${slug}.md`);
      fs.writeFileSync(filename, mdContent);
    });
  } catch (error) {
    console.error('Error fetching data from Notion:', error);
  }
}

fetchNotionPosts();